### PR TITLE
New Notepad_plus_plus Cheat Sheet 

### PR DESCRIPTION
**Pull request by biscuitNinja**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New notepad_plus_plus Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Adds new Notepad++ Cheat Sheet

---

Instant Answer Page: https://duck.co/ia/view/notepad_plus_plus_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @biscuitNinja

### CODE REVIEW

**Comment by daxtheduck** [comment]
@biscuitNinja Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/notepad-plus-plus.json [hunk 1]
```diff
@@ -0,0 +1,286 @@
+{
+    "id": "notepad_plus_plus_cheat_sheet",
+    "name": "Notepad++",
+
+    "metadata": {
+        "sourceName": "notepad-plus-plus.org",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/notepad-plus-plus.json]: Lets have "Notepad plus plus" as source name

File: share/goodie/cheat_sheets/json/notepad-plus-plus.json [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        "sourceUrl" : "https://notepad-plus-plus.org/resources.html"
+    },
+
+    "aliases": [
+        "np++", "notepadplus", "notepad plus", "notepadplusplus", "notepad plus plus"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/notepad-plus-plus.json]: The last alias is same as file name. So we can remove it

File: share/goodie/cheat_sheets/json/notepad-plus-plus.json (hunk 1 above)
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/notepad-plus-plus.json]: I'd do Notepad++ (that's how the product is stylised)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=notepad++ cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/notepad_plus_plus_cheat_sheet_mobile.png?qi1YfO0bAa)

File: share/goodie/cheat_sheets/json/notepad-plus-plus.json (hunk 1 above)
Comment (jophab) [inline share/goodie/cheat_sheets/json/notepad-plus-plus.json]: @mintsoft Yeah, Its good :+1: 
@biscuitNinja Can we have it updated again?

Comment (biscuitNinja) [inline share/goodie/cheat_sheets/json/notepad-plus-plus.json]: @mintsoft @jophab Will amend this eve...

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=notepad++ cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/notepad_plus_plus_cheat_sheet_mobile.png?WGAKTvGcxe)

**Comment by zekiel** [comment]
Nice one @biscuitNinja! Hoping you'll stick around to help w/ the [programming coverage](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage). Overall, looks great and ready for merge! 

A couple things we should note for the future.... 
 1. Looks like the daxtheduck metadata comments don't encode the queries correctly. So if you were to click "notepad++ cheat sheet" in his comment above, it would show you the other Notepad Instant Answer. We'll need to fix that bc it was really confusing me :) 
 2. I found this massive collection of keyboard shortcuts when reviewing your content: https://www.shortcutworld.com/ Certainly useful if you plan to keep tackling these.

**Comment by daxtheduck** [comment]
@biscuitNinja this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**